addons/hr_expense/views/hr_expense_views.xml

```diff
@@ -28,7 +28,10 @@
             <field name="arch" type="xml">
                 <form string="Expenses" class="o_expense_form">
                 <header>
-                    <button name="submit_expenses" states="draft" string="Submit to Manager" type="object" class="oe_highlight o_expense_submit"/>
+                    <button string="Attach Document" special="attachdocument" attrs="{'invisible': [('attachment_number', '&lt;=', 0)]}"/>
```

Comment:
here, I think that the best way to proceed is to use a <widget> and not a special attribute (a form widget is kind of like a field widget, but not linked to a field)